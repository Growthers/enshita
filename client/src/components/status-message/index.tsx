import type { FC } from "react";
import styles from "~/styles/components/status-message/status-message.module.scss";
import type { StatusMessageProps } from "./type/model";

const StatusMessage: FC<StatusMessageProps> = ({ code, message, children }) => (
  <div className={styles["enshita-status-body"]}>
    <div className={styles["enshita-status-content"]}>
      <div className={styles["enshita-status-code"]}>
        <p>{code}</p>
      </div>
      <div className={styles["enshita-status-message"]}>
        <div className={styles["enshita-status-errormessage"]}>
          <p>{message}</p>
        </div>
        <div>{children}</div>
      </div>
    </div>
  </div>
);

export default StatusMessage;
