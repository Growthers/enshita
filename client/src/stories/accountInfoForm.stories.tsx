// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Form from "~/components/form/accountInfoForm";

type T = typeof Form;
type Story = ComponentStoryObj<T>;

export default {
  title: "Account Information Form",
  args: {
    mail: "abc@example.com",
    userName: "hoge",
  },
  component: Form,
} as ComponentMeta<T>;

export const Default: Story = {};
