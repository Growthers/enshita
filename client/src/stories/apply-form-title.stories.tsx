import type { ComponentStoryObj, ComponentMeta } from "@storybook/react";
import { Title } from "~/components/apply-form/title";

type T = typeof Title;
type Story = ComponentStoryObj<T>;

const data = {
  title: "共同開発鯖LT #4",
};

export default { args: { title: data.title }, component: Title } as ComponentMeta<T>;

export const Default: Story = {};
