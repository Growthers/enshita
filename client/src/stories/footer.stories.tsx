// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Footer from "~/components/layout/footer";

type T = typeof Footer;
type Story = ComponentStoryObj<T>;

export default {
  title: "Footer",

  component: Footer,
} as ComponentMeta<T>;

export const Default: Story = {};
