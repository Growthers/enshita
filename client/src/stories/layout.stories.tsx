import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStory } from "@storybook/react";
import Layout from "~/components/layout";

type T = typeof Layout;
type Story = ComponentStory<T>;

export default {
  title: "Layout",

  component: Layout,
} as ComponentMeta<T>;

export const Default: Story = () => (
  <Layout>
    <div>hogehoge</div>
  </Layout>
);
