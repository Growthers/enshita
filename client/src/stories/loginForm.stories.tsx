// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import LoginForm from "~/components/loginForm";

type T = typeof LoginForm;
type Story = ComponentStoryObj<T>;

export default {
  title: "LoginForm",
  component: LoginForm,
} as ComponentMeta<T>;

export const Default: Story = {};
