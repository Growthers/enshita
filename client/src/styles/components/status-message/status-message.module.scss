@use "../../variables" as var;

.enshita-status-body {
  box-sizing: border-box;
  display: flex;
  align-items: center;
  width: 100%;
  height: calc(100vh - 32.5rem);
}

.enshita-status-content {
  display: flex;
  align-items: center;
  justify-content: center;
  width: 100%;
  height: 10rem;
}

.enshita-status-code {
  display: flex;
  align-items: center;
  width: 20rem;
  height: 100%;
  border-color: var.$global-black;
  border-right: 2px solid;

  p {
    font-size: 10rem;
  }
}

.enshita-status-message {
  width: 20rem;
  height: 100%;
  padding-left: 1rem;
  border-left: 2px solid;

  .enshita-status-errormessage {
    font-size: 2rem;
  }

  div {
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    height: 50%;
    text-align: center;
  }
}
