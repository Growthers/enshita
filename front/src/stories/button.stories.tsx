import React from "react";
// eslint-disable-next-line import/no-extraneous-dependencies
import { ComponentMeta, ComponentStoryObj } from "@storybook/react";
import Button from "~/components/button";

type T = typeof Button;
type Story = ComponentStoryObj<T>;

export default {
  title: "Button",
  args: { children: <div>hogehoge</div> },
  component: Button,
} as ComponentMeta<T>;

export const Default: Story = {};
