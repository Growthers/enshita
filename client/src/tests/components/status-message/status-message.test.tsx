import { composeStories } from "@storybook/testing-react";
import "@testing-library/jest-dom";
import { render } from "@testing-library/react";
import * as stories from "~/stories/status-message/status-message.stories";

const { Default } = composeStories(stories);

describe("(components) header", () => {
  test("test Snap Shot", () => {
    const { container } = render(<Default />);
    expect(container).toMatchSnapshot();
  });
});
