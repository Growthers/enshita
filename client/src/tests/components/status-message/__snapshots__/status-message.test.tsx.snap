// Vitest Snapshot v1

exports[`(components) header > test Snap Shot 1`] = `
<div>
  <div
    class="_enshita-status-body_3a3ce3"
  >
    <div
      class="_enshita-status-content_3a3ce3"
    >
      <div
        class="_enshita-status-code_3a3ce3"
      >
        <p>
          404
        </p>
      </div>
      <div
        class="_enshita-status-message_3a3ce3"
      >
        <div
          class="_enshita-status-errormessage_3a3ce3"
        >
          <p>
            page not found
          </p>
        </div>
        <div>
          <p>
            Oh... This URL is wrong.
            <br />
            please check again page URL.
          </p>
        </div>
      </div>
    </div>
  </div>
</div>
`;
